blockprotocol/blockprotocol
# Notify Slack upon block publication/update on the Hub

## What is the purpose of this PR?

This PR sends notifications to an internal slack channel when blocks on Block Hub gets updated, or new ones gets created.

## What does this change?

- Adds `SLACK_BLOCK_NOTIFICATION_WEBHOOK_URL` env variable to vercel/blockprotocol project
- Calls new `notifySlackAboutBlock` function when `api/blocks/create` or `api/blocks/publish` endpoints succeeds

## How to test this?

1. Check an [example slack notification](https://hashintel.slack.com/archives/C042NKNA2EN/p1663301969710919) _(internal)_
**OR**
1. Checkout to the branch
2. Uncomment this line: https://github.com/blockprotocol/blockprotocol/blob/24c985acfd83b6528d259783ff4b56526e6faeb3/site/src/lib/api/blocks/slack.ts#L11
3. Try publishing via local BP CLI, or http://localhost:3000/blocks/publish/npm

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202961851742601
